### PR TITLE
extensions: log extension activation events from extension host

### PR DESCRIPTION
**Pull request by tjkandala**
We stopped logging extension activation events after the extension host refactor in #18898 

- Add `logEvent` to main thread API so we can track events from the extension host
- Log extension activation events through `MainThreadAPI#logEvent`

### CODE REVIEW

**Review by Joelkw — APPROVED** [review]
PR description and a glance at these changes looks good to me, but noting that I'm not very with these areas of our codebase / haven't had too much time to dig into how the refactor works so approving this with those caveats. One way we should be able to get quick additional confidence that this change works is that it should immediately restore our extension pings from dotcom (you can check that here https://sourcegraph.looker.com/looks/772), though, so I'd advocate for merging this tomorrow if @muratsu has a chance to review.

**Comment by tjkandala** [comment]
@muratsu I've added tests for `activation.ts`, the first of which asserts that we log extension activation (check latest commit)

File: client/shared/src/api/extension/activation.ts [hunk 1]
```diff
@@ -134,23 +143,44 @@ export function activateExtensions(
...
+                                    // We only want to log non-default extension events
+                                    if (!defaultExtensions[id]) {
+                                        // Hash extension IDs that specify host, since that means that it's a private registry extension.
+                                        const telemetryExtensionID = splitExtensionID(id).host ? hashCode(id, 20) : id
+                                        mainAPI
+                                            .logEvent('ExtensionActivation', {
+                                                extension_id: telemetryExtensionID,
+                                            })
+                                            .catch(() => {
+                                                // noop
```
Comment (muratsu) [inline client/shared/src/api/extension/activation.ts:168]: Would logging this be useful for debugging potential telemetry exceptions in the future? Say if we ask for logs from clients?

**Review by muratsu — APPROVED** [review]

**Comment by muratsu** [comment]
LGTM

**Pull request merged**